worker-document-schema/src/main/resources-yaml/com/microfocus/caf/worker/document/schema/model/schema.yaml

```diff
@@ -0,0 +1,167 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://cafdataprocessing.github.io/worker-document/schema.json
+title: Document worker wocument schema
+description: The schema for a document that a Document Worker can process.
+type: object   
+properties: 
+  document:
+    "$ref": "#/definitions/workerDocument"
+  changeLog:
+    "$ref": "#/definitions/changeLog"
+additionalProperties: false
+definitions:
+  workerDocument:
+    type: object
+    properties:
+      reference:
+        type: string
+      fields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            "$ref": "#/definitions/fieldValue"
+      failures:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/failure"
+      subdocuments:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/workerDocument"
+    additionalProperties: false
+  fieldValue:
+    type: object
+    required:
+    - data
+    properties:
+      data:
+        type: string
+      encoding:
+        type: string
+        default: utf8
+        enum: [utf8, base64, storage_ref, local_ref]
+    additionalProperties: false
+  failure:
+    type: object
+    properties:
+      failureId:
+        type: string
+      failureMessage:
+        type: string
+      failureStack:
+        type: string
+    additionalProperties: false
+  changeLog:
+    type: array
+    items:
+      type: object
+      "$ref": "#/definitions/changeLogEntry"
+  changeLogEntry:
+    type: object
+    properties:
+      name:
+        type: string
+      changes:
+        type: array
+        items:
+          type: object
+          "$ref": "#/definitions/change"
+    additionalProperties: false
+  change:
+    type: object
+    oneOf:
+    - required: [setReference]
+    - required: [addFields]
+    - required: [setFields]
+    - required: [addFailure]
+    - required: [addSubdocument]
+    - required: [insertSubdocument]
+    - required: [updateSubdocument]
+    - required: [removeSubdocument]
+    properties:
+      setReference:
+        type: object
+        properties:
+          value:
+            type: string
+        additionalProperties: false
+      addFields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            type: object
+            "$ref": "#/definitions/fieldValue"
+      setFields:
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            type: object
+            "$ref": "#/definitions/fieldValue"
+      removeFields:
+        type: array
+        items:
+          type: string
+      addFailure:
+        type: object
+        "$ref": "#/definitions/failure"
+      setFailures:
+        type: array
+        items:
+          type: object
```

Comment:
I don't think this `type: object` line should be present (since it comes from the referenced definition).

Reply:
removed type: object